### PR TITLE
Calendar Date Selection Redesign

### PR DESCRIPTION
**Pull request by nmakuch**
### Date Buckets
- Changing Date Bucket Design (Mobile and Desktop).
- Date Buckets are invisible until a date has been selected.
- Error Message design inside of buckets
- Mobile Styling of Buckets

| Before | After |
|--------|-------|
|  buckets-before|   buckets-after|

### Footer Glitch Fix
- Noticed a weird spacing in the Footer caused by the gcui element, removed it and created
- This glitch had been appearing on all pages in mobile view
- Updated the footer tests 
- Look is identical

### Calendar Redesign
- Disabled dates/weekday headings BG Gray, black text, regular weight
- Selectable days/weekday headings BG white, black text, bold weight
- Testing mobile Calendar on iOS and Android devices

| Before | After |
|--------|-------|
| calendar-before| calendar-after|

## Design Changes left out (Discussed with Eman):
- Going with squares instead of circles, we want the width of the calendar to be able to be liquid and this is easier with squares

# This PR will close [ISSUE 191](https://github.com/cds-snc/ircc-rescheduler/issues/191)

### CODE REVIEW

File: web/src/assets/CanadaWordmark.svg [hunk 1]
```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Generator: Adobe Illustrator 22.1.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
```
Comment (timarney) [inline web/src/assets/CanadaWordmark.svg]: I would suggest running this SVG through 
https://jakearchibald.github.io/svgomg/

Should clean up all the unneeded Adobe comments etc... + will drop the files size a little (worth a try IMO)

File: web/src/assets/cancel.svg [hunk 2]
```diff
@@ -1 +1,17 @@
-<svg id="Layer_1" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 22.63 22.63"><defs><style>.cls-1{fill:#fff;}.cls-2{stroke:#000;stroke-miterlimit:10;stroke-width:0.3px;}</style></defs><title>cancel</title><circle class="cls-1" cx="11.31" cy="11.31" r="10.81"/><path d="M16.11,16.64a.42.42,0,0,1-.3-.12L7.6,8.31a.41.41,0,0,1,0-.58.42.42,0,0,1,.59,0l8.21,8.21a.41.41,0,0,1,0,.58A.42.42,0,0,1,16.11,16.64Z" transform="translate(-0.69 -0.81)"/><path d="M7.89,16.64a.42.42,0,0,1-.29-.12.41.41,0,0,1,0-.58l8.21-8.21a.42.42,0,0,1,.59,0,.41.41,0,0,1,0,.58L8.19,16.52A.42.42,0,0,1,7.89,16.64Z" transform="translate(-0.69 -0.81)"/><path class="cls-2" d="M12,23.29A11.17,11.17,0,1,1,23.16,12.12,11.17,11.17,0,0,1,12,23.29Zm0-21.5A10.34,10.34,0,1,0,22.34,12.12,10.35,10.35,0,0,0,12,1.79Z" transform="translate(-0.69 -0.81)"/></svg>
```
Comment (timarney) [inline web/src/assets/cancel.svg]: Would suggest the same for all the SVGs https://jakearchibald.github.io/svgomg

File: web/src/components/ErrorMessage.js [hunk 3]
```diff
@@ -53,7 +54,7 @@ class ErrorList extends React.Component {
 
   render() {
     return (
-      <div className={this.isEmpty() ? `empty ${noErrorList}` : errorList}>
```
Comment (timarney) [inline web/src/components/ErrorMessage.js]: Should be able to do something like {this.isEmpty ? null : errorList} here to save the css work ... might make it a little easier to see what's going on.  Also won't render a hidden div if it's not needed (not sure what is better on the a11y end - hidden item or item rendering an item after the initial load).

File: web/src/components/Calendar.js [hunk 4]
```diff
@@ -343,40 +370,107 @@ const dayBox = css`
 `
 
 const daySelection = css`
+  background: ${theme.colour.blackLight} !important;
   margin-bottom: ${theme.spacing.xl};
+  width: 20rem !important;
+  padding: ${theme.spacing.lg} ${theme.spacing.lg} 0 ${theme.spacing.lg};
+
+  li {
+    border-top: 1px solid #dddddd;
```
Comment (pcraig3) [inline web/src/components/Calendar.js]: Another colour here. Might end up being nuts trying to put them all into `styles.js`, so maybe let's hold off if this is the only place. But if we start re-using them, then yes.

File: web/src/components/Calendar.js [hunk 5]
```diff
@@ -343,40 +370,107 @@ const dayBox = css`
...
+  height: 0;
+  margin-top: ${theme.spacing.lg};
+  margin-left: ${theme.spacing.md};
+  border-top: 20px solid transparent;
+  border-bottom: 20px solid transparent;
+  border-right: 25px solid ${theme.colour.blackLight};
+
+  ${mediaQuery.lg(css`
+    display: none;
+  `)};
```
Comment (pcraig3) [inline web/src/components/Calendar.js]: This is super cool. Neat CSS trick. 👍

File: web/src/components/Calendar.js [hunk 6]
```diff
@@ -550,12 +640,11 @@ class Calendar extends Component {
           onBlur={() => onBlur(value)}
           containerProps={{ id, tabIndex }}
         />
-        <div className={daySelection}>
-          <h3>
-            <Trans>Your 3 selected days:</Trans>
-          </h3>
+        <div className={value.length ? triangle : noDates} />
+        <div className={value.length ? daySelection : noDates}>
+          <h3>Your 3 selected days:</h3>
```
Comment (pcraig3) [inline web/src/components/Calendar.js]: "Your 3 selected days:" should be inside of a `<Trans>` component.

File: web/src/components/Calendar.js [hunk 7]
```diff
@@ -300,28 +331,24 @@ const calendarContainer = css`
...
+      font-size: ${theme.font.md};
     `)};
 
     &.empty {
-      background-color: ${theme.colour.greyLight};
-      border: 2px solid ${theme.colour.greyLight};
-
       * {
-        visibility: hidden;
+        display: none;
```
Comment (pcraig3) [inline web/src/components/Calendar.js]: I was looking into this, whether setting an error message from `display:none` to `display:block` was a good idea.

I haven't found what looks like the definitive answer.

- [this article](http://web-accessibility.carnegiemuseums.org/code/alerts/) says to use `display:none` to hide error messages until we need them
- [this article](http://www.davidmacd.com/blog/test-aria-live-display-none.html) and [this one](https://developer.paciellogroup.com/blog/2012/06/html5-accessibility-chops-aria-rolealert-browser-support/) are both saying that it might not work on Safari

Let's keep your CSS but then we'll run this manually / see if the accessibility review picks this up.

File: web/src/components/ErrorMessage.js (hunk 3 above)
Comment (pcraig3) [reply web/src/components/ErrorMessage.js]: i. the calendar component is looking for an `.empty` class to apply some CSS. Also, the test I wrote for this is looking for an `empty` class because I was trying to test that the logic was working properly.  Not sure it's the ideal solution, but for now would prefer to leave it in.

ii. Looked this up, and it looks like we want the empty div to be on the page when it loads.
- according to [some W3C aria guidance](https://www.w3.org/TR/WCAG20-TECHS/ARIA19.html?f5tp=1), it says "The error container must be present in the DOM on page load for the error message to be spoken by most screen readers."
- and then on [this scott vinkle blog post](https://medium.com/@svinkle/creating-accessible-react-apps-38dc1c2775cf), he says "Additionally, it’s important to always render the Announcement component in the template as some browser/screen reader technologies will not announce content when the aria-live element is dynamically added to the DOM. As a result, this component should always be included in any parent components in your app."

I was looking into whether `display:none` causes a problem here as well (see above comment), but I think that's probably okay.

File: web/src/components/ErrorMessage.js [hunk 8]
```diff
@@ -86,7 +87,9 @@ class ValidationMessage extends React.Component {
   render() {
     return (
       <span
-        className={`${this.props.message ? '' : 'empty '}${errorMessage}`}
+        className={`${
+          this.props.message ? '' : `empty ${noError}`
+        }${errorMessage}`}
```
Comment (pcraig3) [inline web/src/components/ErrorMessage.js]: This could also be 
```
className={this.props.message ? errorMessage : `empty ${noError}`}
```

File: web/src/components/ErrorMessage.js [hunk 9]
```diff
@@ -107,10 +110,19 @@ ValidationMessage.propTypes = {
...
+
+  isEmpty() {
+    return !(this.props.message ? true : false)
+  }
+
   render() {
     return (
       <span
-        className={`${this.props.message ? '' : 'empty '}${errorMessage}`}
+        className={this.isEmpty() ? `empty ${noError}` : errorMessage}
```
Comment (pcraig3) [inline web/src/components/ErrorMessage.js]: since `isEmpty` is super simple here, we could just use the check for the message.

Like this
```
 className={!(this.props.message) ? `empty ${noError}` : errorMessage}
```

And then we could remove the constructor and the `isEmpty` function.

Comment (pcraig3) [reply web/src/components/ErrorMessage.js]: Actually, it could be exactly the same as the comment before ☝️ 

```
className={this.props.message ? errorMessage : `empty ${noError}`}
```

File: web/src/components/Footer.js [hunk 10]
```diff
@@ -116,12 +130,18 @@ const Footer = withI18n()(({ topBarBackground, i18n, context = {} }) => (
...
+          alt={
+            context.store.language === 'en'
+              ? 'Symbol of the Government of Canada'
+              : 'Symbole du gouvernement du Canada'
+          }
+          aria-label={
+            context.store.language === 'en'
+              ? 'Symbol of the Government of Canada'
+              : 'Symbole du gouvernement du Canada'
+          }
```
Comment (pcraig3) [inline web/src/components/Footer.js]: Pretty sure we only want one of these, either `alt` or `aria-label`. In this case, I think let's go with `alt`.

File: web/src/components/__tests__/Footer.test.js [hunk 11]
```diff
@@ -38,4 +38,14 @@ describe('<Footer />', () => {
...
   })
+
+  it('renders with Canadawordmark in French', () => {
+    const footer = mount(<Footer context={getStore('fr')} />)
+    expect(footer.find('img').length).toBe(1)
+  })
+
+  it('renders with Canadawordmark in English', () => {
+    const footer = mount(<Footer context={getStore('en')} />)
+    expect(footer.find('img').length).toBe(1)
```
Comment (pcraig3) [inline web/src/components/__tests__/Footer.test.js]: You should also get the "alt" attribute of the image and check that. There's definitely a way using enzyme, maybe [check here](https://github.com/airbnb/enzyme/issues/669#issuecomment-260086809). If you can't figure it out, let me know.

File: web/src/components/Calendar.js (hunk 4 above)
Comment (pcraig3) [reply web/src/components/Calendar.js]: Actually, I've just checked and we are using `#DBDBDB`, which is pretty much the same. Could swap this one for the one we already have.
